jwst/ramp_fitting/ramp_fit.py

```diff
@@ -175,23 +176,25 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
             nreads -= 1
             ngroups -= 1
             cubeshape = (nreads,)+imshape
-            log.info('MIRI dataset has all final reads flagged as DO_NOT_USE.')
+            log.info('MIRI dataset has all final groups flagged as DO_NOT_USE.')
 
-    if (ngroups == 1):
-        log.warn('Dataset has NGROUPS=1, so count rates for each integration')
-        log.warn('will be calculated as the value of that 1 group divided by')
-        log.warn('the group exposure time.')
+        # Next block is to satisfy github issue 1681:
+        # "MIRI FirstFrame and LastFrame minimum number of groups #1681"
+        if (ngroups < 2):
+            log.error('MIRI datasets require at least 2 groups/integration')
+            log.error('(NGROUPS), so will not process this dataset.')
+            return None, None, None
 
-    slopes = np.zeros(imshape, dtype=np.float32)
+    if (ngroups == 1):
+        log.warning('Dataset has NGROUPS=1, so count rates for each integration')
+        log.warning('will be calculated as the value of that 1 group divided by')
+        log.warning('the group exposure time.')
 
-    slope_wtd = np.zeros(imshape, dtype=np.float64)
-    m_sum_2d = np.zeros(imshape, dtype=np.float64)
-    var_sum_2d = np.zeros(imshape, dtype=np.float64)
+    med_slopes = np.zeros(imshape, dtype=np.float32)
+    slope_wtd = np.zeros(imshape, dtype=np.float32)
```

Comment:
It doesn't look as if `slope_wtd` is ever used.

Reply:
You are correct - updating.